python/src/equistore/operations/slice.py

```diff
@@ -0,0 +1,301 @@
+        or each of the sliced :py:class:`TensorBlock` objects of the output
+        :py:class:`TensorMap`. Default value of None indicates no slicing along the
+        samples dimension should occur.
+    :param properties_to_slice: a :py:class:`Labels` object containing the names and
+        indices of properties to keep in the sliced :py:class:`TensorBlock` output,
+        or each of the sliced :py:class:`TensorBlock` objects of the output
+        :py:class:`TensorMap`. Default value of None indicates no slicing along the
+        properties dimension should occur.
+
+    :return: a :py:class:`TensorBlock` or :py:class:`TensorMap` that corresponds
+        to the sliced input.
+
+    .. code-block:: python
+            sliced_tensor = slice(
+                tensor,
+                samples=Labels(
+                    names=["structure", "center"],
+                    values=np.array(
+                        [[0, 1], [0, 6], [1, 6], [3, 16]]
+                    ),  # must be a 2D-array
+                )
+                properties=Labels(
+                    names=["n",],  # radial channel
+                    values=np.array([[3,], [4,], [5,]]),
+                )
+            )
+    """
+
+    # Perform input check on the tensor
+    if not isinstance(tensor, (TensorBlock, TensorMap)):
+        raise ValueError(
+            "Input `tensor` must either be an equistore TensorBlock or TensorMap."
+        )
+
+    # Perform some checks on the passed args `samples_to_slice`
+    # and `properties_to_slice`.
+    if samples_to_slice is None and properties_to_slice is None:
+        raise ValueError(
+            "Must specify either some samples or properties (or both) to slice by."
+        )
+    if samples_to_slice is not None:
+        assert isinstance(
+            samples_to_slice, Labels
+        ), "samples_to_slice must be passed as a equistore.Labels object."
+        sample_names = (
+            tensor.sample_names
+            if isinstance(tensor, TensorMap)
+            else tensor.samples.names
+        )
+        for name in samples_to_slice.names:
+            assert name in sample_names, (
+                "Invalid sample name '" + name + "' - doesn't exist in input tensor."
+            )
+    if properties_to_slice is not None:
+        assert isinstance(
+            properties_to_slice, Labels
+        ), "properties_to_slice must be passed as a equistore.Labels object."
+        property_names = (
+            tensor.property_names
+            if isinstance(tensor, TensorMap)
+            else tensor.properties.names
+        )
+        for name in properties_to_slice.names:
+            assert name in property_names, (
+                "Invalid property name '" + name + "' - doesn't exist in input tensor."
+            )
+
+    # Slice tensor and issue warnings if some or all blocks are empty
+    warnings.filterwarnings(action="always")
```

Comment:
Why is this here? We should not be changing the global warnings configuration from inside a library function

Reply:
Removed in commit f49fe0b215ca61158597a18c8eb92fffacbac168.